CONTRIBUTING.md

```diff
@@ -31,25 +40,57 @@ You can read more about it at their [official documentation](https://docs.github
-
-For documentation changes to show up at <https://obsidian-tasks-group.github.io/obsidian-tasks/> , they must be in the `gh-pages` branch.
-If you want to see your changes available immediately and not only after the next release, you should make your changes on the `gh-pages` branch.
-When you create a PR, it should merge into the `gh-pages` branch as well.
-If you document an unreleased feature, you should update the documentation on `main` instead. Ideally together with the related code changes.
-If this is confusing, don't worry.
-We will help you make this right once you opened the PR.
-
 ## Updating code
 
-Ideally, an [issue](https://github.com/obsidian-tasks-group/obsidian-tasks/issues) already exists and we discussed your implementation in that issue before you open the pull request (PR).
+Ideally, an [issue](https://github.com/obsidian-tasks-group/obsidian-tasks/issues) or
+[Discussion](https://github.com/obsidian-tasks-group/obsidian-tasks/discussions) already exists
+and we discussed your implementation in that issue before you open the pull request (PR).
 This is _not_ mandatory, but it helps improve the process and reduce unnecessary back-and-forth.
 
 Once you want to propose your changes, create a PR and we'll have a look when we have time.
 Discussion will take place inside the PR.
 
 If you can, please add/update tests and documentation where appropriate.
 
+## Local setup and workflow for changes to code and tests
+
+This project uses the Yarn package manager for Javascript, and several formatting and linting tools.
+Below are specific setup instructions for changing code and tests, as well as tips for local development workflow.
+
+### Setting up build environment
+
+This project uses Node 14.x, if you also use a different version, look at using `nvm` to manage your Node versions.
+If you are using `nvm`, you can install the 14.x version of Node with `nvm install 14; nvm use <full version number you installed such as 14.19.3>`.
+
+To setup the local environment after cloning the repository, run the following commands:
+
+```shell
+yarn
+yarn build
+yarn test
+yarn lint
+yarn lint:markdown
+```
+
+Make sure you build, test and lint before pushing to the repository. Lefthook is used to cover these checks before commit and push.
+
+### Local development
+
+When developing locally, you can use the `yarn dev` command to start a development build.
+This will cause a rebuild of the code base every time you make a change so you can see if there are any code errors.
+
+Not all the functionality of Tasks can be tested via the automated tests. If you want to test your changes in a local obsidian vault use `yarn run build:dev`.
+This will generate the `main.js` in the root of the repository with a sourcemap in it to facilitate
+debugging using the development console (`Ctrl+Shift+i` on Windows or `Cmd+Shift+i` on Mac) in Obsidian.
+
+If your environment supports Powershell scripts (e.g. Windows), you can use `yarn deploy:local`
```

Comment:
PowerShell has been cross-platform for some time:
https://devblogs.microsoft.com/dotnet/powershell-is-now-open-source-and-cross-platform/